src/harness/util.py

```diff
@@ -425,4 +426,94 @@ def compareDictWithUnorderedLists(first_dict, second_dict):
     second_dict: second dictionary to be compared.
  Returns: boolean set to true if the dictionaries are equal and false otherwise.
   """
-  return _orderAttributes(first_dict) == _orderAttributes(second_dict)
\ No newline at end of file
+  return _orderAttributes(first_dict) == _orderAttributes(second_dict)
+
+def _areTwoPointsEqual(first_point, second_point, delta):
+  """ a comparison of two points
+
+  Args:
+    first_point: an array which is the coordinates of the first point.
+
+    second_point:  an array which is the coordinates of the second point.
+    delta: an accepted margin for comparing the points
+ Return:  boolean set to true if the points are equal and false otherwise """
+  return abs(first_point[0] - second_point[0]) <= delta \
+    and abs(first_point[1] - second_point[1]) <= delta
+
+
+def _areTwoContoursEqual(first_contour, second_contour, delta):
```

Comment:
Why not just do polygon comparisons like the ones in the PCR test cases? That seems to achieve the same goal while being less brittle (I think a SAS would fail this code if they e.g. inserted a third point exactly between two existing points, even though it doesn't change the shape of the polygon).

Reply:
do you mean the Equation 8.3-1 of test spec of PCR?
if yes , in FAD test  cases the data are not calculated but they are injected in SAS UUT, so why will SAS UUT change them by adding points/holes or deleting points/holes(except of having a small difference in the coordinates of the points because of the decimal point presentation)?

Reply:
Yes, I mean that equation. A SAS may convert to another format for polygon representation and then back to JSON for export, and the conversion could add (or remove) vertices which do not change the shape of the polygon.

In the end, I don't think that we care so much that the vertices match the injected PPA exactly -- after all, we will not be injecting PPAs during normal operation -- and we care more that it accurately represents the area covered by the PPA. To me, polygon comparison gives the right amount of flexibility: it checks exactly the requirement we wish to impose on the SAS, and doesn't impose additional requirements.

Reply:
Hi, I have created the utility routine for comparing the PPA, reusing the work done on polygon area and using the methodology of Eq 8.3.1.

Please see at PR #477

Reply:
By the way, please go in PR #477 and review that code so I can submit it.

Reply:
thank you Serge for this useful function, I approved it and I used it in this test, so please go ahead and submit it.

by the way I noticed that the function from GeoJsonToShapelyGeometry function doesn't keep the same order of the points of the linearRings in the polygon.

Reply:
Sure. 
The function GeoJsonToShapelyGeometry uses the shapely shape() routine which adapts any object implementing the  __geo_interface__ (see https://gist.github.com/sgillies/2217756), which is itself defined as a geojson-like dict. So I would assume that it would keep the dict of the input as is.
Then it passes it to the buffer(0) call, whose purpose is only to fix invalid polygons. No reason for it to modify order of points, unless if the polygon is invalid.
So I am a bit surprised than one of these 2 calls would reshuffle the linear ring.

But in any case, I dont think this is an issue, any function working on polygon should not assume anything about the linear ring points order or even direction (CW or CCW): for example 2 polygons can be completely equal with differently ordered exterior or interior contours.

If this is an issue for something in your opinion, let me know and we can analyze further.

Reply:
I think in this test case(FAD.1) we need to check that the polygon returned has CCW orientation, but I saw that the points of polygon returned by GeoJsonToShapelyGeometry are not ccw while the points of original polygon are ccw,
please see below the example I used which show that the order is not the same, (sorry for using too many digit after the decimal point :( )

the GeoJSON polygon exteriour ring:

[[  -97.23861694335938,  38.865374851611634],[  -97.31964111328125,  38.7615795117574],[  -97.16995239257811,  38.739088441876866],[  -97.16171264648436,  38.86751337001198],[  -97.23861694335938,  38.865374851611634]]
the linearRing of its shapely polygon:

LINEARRING (-97.23861694335938 38.86537485161163, -97.16171264648436 38.86751337001198, -97.16995239257811 38.73908844187687, -97.31964111328125 38.7615795117574, -97.23861694335938 38.86537485161163)